lib/hydra/works/services/generic_file/persist_derivative.rb

```diff
@@ -9,17 +9,16 @@ class PersistDerivative < Hydra::Derivatives::PersistOutputFileService
     # The purpose of this Service is for use as an alternative to the default Hydra::Derivatives::PersistOutputFileService.  It's necessary because the default behavior in Hydra::Derivatives assumes that you're using LDP Basic Containment. Hydra::Works::GenericFiles use IndirectContainment.  This Service handles that case.
     # 
     # @param [Hydra::Works::GenericFile::Base] object the file will be added to
-    # @param [File] file the dervivative filestream
-    # @param [String] destination_name path to the file
+    # @param [#read] file the dervivative filestream optionally responds to :mime_type
     # @option opts [RDF::URI or String] type URI for the RDF.type that identifies the file's role within the generic_file
     # @option opts [Boolean] update_existing when set to false, always adds a new file.  When set to true, performs a create_or_update
     # @option opts [Boolean] versioning whether to create new version entries
 
-    def self.call(object, file, destination_name, opts={})
+    def self.call(object, file, opts={})
       type = opts.fetch(:type, ::RDF::URI("http://pcdm.org/use#ServiceFile"))
       update_existing = opts.fetch(:update_existing, true)
       versioning = opts.fetch(:versioning, true)
-      Hydra::Works::AddFileToGenericFile.call(object, destination_name, type, update_existing: update_existing, versioning: versioning)
+      Hydra::Works::AddFileToGenericFile.call(object, file, type, update_existing: update_existing, versioning: versioning)
```

Comment:
Note: the way Hydra::Derivatives is set up right now, `file` is a string of bytestream content and opts[:mime_type] is set.  This brings up 2 things:

1. that code in hydra-derivatives should be refactored to pass a file object that responds to mime_type and original_filename
2. this code in hydra-works should provide some example (possibly in the tests?) of how to wrap an IO object or a string in an object that responds to `.mime_type` and `.original_filename`, so we could just do that in Hydra::Works::PersistDerivative.call for now

See https://github.com/projecthydra/hydra-derivatives/blob/7a8377c12ba1697f825cef9f5b2cd3ec7b844228/lib/hydra/derivatives/document.rb#L27